scoring/src/main/scala/ai/h2o/sparkling/ml/params/H2OTargetEncoderParams.scala

```diff
@@ -75,7 +77,14 @@ trait H2OTargetEncoderParams extends Params {
 
   def getInputCols(): Array[String] = $(inputCols)
 
-  def getOutputCols(): Array[String] = getInputCols().map(_ + "_te")
+  def getOutputCols(): Array[String] = {
+    val columns = $(outputCols)
+    if (columns == null || columns.isEmpty) {
```

Comment:
`outputCols` is `StringArrayParam` and it is not nullable, `null` should not be a valid option. I think we might just prevent user setting null, what do you thing?

Reply:
Or we can keep the option to specify `null` but the parameter needs to be changed to `NullableStringArrayParam`

Reply:
Yes, you are right, `null` is not a valid option. `columns == null` is extra